sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -1070,15 +1070,32 @@ public static RetryConfiguration create(
      *
      * <pre>{@code
      * PCollection<Void> firstWriteResults = data.apply(JdbcIO.write()
-     *     .withDataSourceConfiguration(CONF_DB_1).withResults());
+     *     .withDataSourceConfiguration(CONF_DB_1).withVoidResults());
      * data.apply(Wait.on(firstWriteResults))
      *     .apply(JdbcIO.write().withDataSourceConfiguration(CONF_DB_2));
      * }</pre>
      */
-    public WriteVoid<T> withResults() {
+    public WriteVoid<T> withVoidResults() {
       return inner;
     }
 
+    /**
+     * Returns {@link WriteWithResults} transform that could return a specific result.
+     *
+     * <p>See {@link WriteWithResults}
+     */
+    public <V> WriteWithResults<T, V> withReturningResults(RowMapper<V> rowMapper) {
+      return new AutoValue_JdbcIO_WriteWithResults.Builder<T, V>()
```

Comment:
To avoid copying the same config methods, it might make sense to move them into separate class, e.g. JdbcConfig (like [BigtableConfig](https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfig.java)). Though, we can leave it for refactoring later.